scripts/lib/CIME/case/case.py

```diff
@@ -1123,6 +1128,23 @@ def configure(self, compset_name, grid_name, machine_name=None,
         if test:
             self.set_value("TEST",True)
 
+        # sanity check: when use pgi-gpu and nvhpc-gpu compiler, we must have
+        #               at least one gpu per node available
+        if compiler in ["pgi-gpu", "nvhpc-gpu"]:
+            expect(ngpus_per_node > 0," ngpus_per_node is expected > 0 for compiler {}; current value is {}".format(compiler, ngpus_per_node))
+        else:
+            expect(ngpus_per_node == 0," ngpus_per_node is expected = 0 for compiler {}; current value is {}".format(compiler, ngpus_per_node))
```

Comment:
Thanks for the comment. The default value for `ngpus_per_node` is 0. I think the sanity check here is necessary as it could avoid some invalid values provided by the user (e.g., >0 value for a machine without GPU and <0 by mistake). I agree that it is better to make it more generic but it will be helpful if there is an agreement about the name of compiler that supports GPU run (e.g., always ending with `gpu`?).

Reply:
There has never been a standard in CIME for making the same compiler have the same name across machine config entries.  Instead of hard-coding the string, can you look for "gpu" in the compiler name which is pretty common?

Reply:
Thanks @rljacob. That is a good suggestion and I will update the if-condition here accordingly.